src/inputs/TextField/TextField.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import styles from './TextField.module.css'
+import { h } from 'preact'
+import { Fragment } from 'react'
+import { Delete } from '../../assets/Icons/Delete'
+import { Error } from '../../assets/Icons/Error'
+
+interface Props {
+	autoFocus?: boolean
+	disabled?: boolean
+	value: string
+	name: string
+	placeholder: string
+	error?: string
+	hint?: string
+	onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
+	label: string
+	id: string
+	type?: string
+	icon?: React.ReactNode
+}
+
+const TextField: React.FC<Props> = ({
+	autoFocus = false,
+	disabled = false,
+	label,
+	value,
+	name,
+	placeholder,
+	error,
+	hint,
+	onChange,
+	id,
+	type = 'text',
+	icon,
+	...props
+}) => {
+	return (
+		<>
+			<input
+				className={`${styles.input} ${autoFocus && styles.inputFocus} ${error && styles.inputError}  ${
```

Comment:
**suggestion:** I would move this class to a const, because it's a lot of logic here. It will be difficult to maintain if you add more things